src/main/java/org/mockito/internal/stubbing/answers/InvocationInfo.java

```diff
@@ -25,14 +26,33 @@ public InvocationInfo(InvocationOnMock theInvocation) {
     }
 
     public boolean isValidException(Throwable throwable) {
+        if (isValidException(method, throwable)) {
+            return true;
+        }
+
+        return Arrays.stream(method.getDeclaringClass().getInterfaces())
+                .anyMatch(parent -> isValidExceptionForParent(parent, throwable));
```

Comment:
I am not 100% sure, but I think we should also recursively walk up the parent class hierarchy in case the direct parent doesn't define the method. E.g. I think the current regression tests still fails:

```java
public interface Foo {
    void bar() throws IOException; 
}
public abstract class Baz implements Foo {}
public class SubClass extends Baz {
    @Override
    public void bar() {
    }
}
```